src/pages/federation.tsx

```diff
@@ -160,4 +243,202 @@ data.post("/refresh", async (c) => {
   return c.redirect("/federation?error=refresh");
 });
 
+data.post("/relay", async (c) => {
```

Comment:
I'd be inclined to move this to a dedicated `relays` page, as this file already is quite large.